src/detection_linux.cpp

```diff
@@ -229,35 +206,70 @@ void DeviceRemoved(struct udev_device* dev) {
 	currentItem = item;
 	isAdded = false;
 
-	SignalDeviceAvailable();
+	uv_async_send(&async_handler);
 }
 
 
-void* ThreadFunc(void* ptr) {
-	while (1) {
-		/* Make the call to receive the device.
-		   select() ensured that this will not block. */
-		dev = udev_monitor_receive_device(mon);
-		if (dev) {
-			if(udev_device_get_devtype(dev) && strcmp(udev_device_get_devtype(dev), DEVICE_TYPE_DEVICE) == 0) {
-				if(strcmp(udev_device_get_action(dev), DEVICE_ACTION_ADDED) == 0) {
-					WaitForDeviceHandled();
-					DeviceAdded(dev);
-				}
-				else if(strcmp(udev_device_get_action(dev), DEVICE_ACTION_REMOVED) == 0) {
-					WaitForDeviceHandled();
-					DeviceRemoved(dev);
+static void cbWork(uv_work_t *req) {
+	// We have this check in case we `Stop` before this thread starts,
+	// otherwise the process will hang
+	if (isRunning) {
```

Comment:
For code style consistency, and to reduce one level of indentation on this function body, this could be re-written as:
```cpp
if (!isRunning) {
    return;
}

// Rest of function body...
```